docs/advanced/custom-directive.md

```diff
@@ -0,0 +1,73 @@
+---
+id: custom-directive
+pagination_prev: null
+pagination_next: null
+---
+
+# Custom schema directives handling
+
+For directives applied to the schema, you can select which ones to be rendered for the types or in the locations they are declared. Information about the custom directives includes a custom description.
+
+For example, we have one query called `searchRole`, and we want to limit access to `ADMIN` user roles only.
+
+We can accomplish this by adding a directive called `auth` with an argument `requires` to the query.
+
+```graphql
+directive @auth(
+  requires: Roles = ADMIN
+) on OBJECT | FIELD_DEFINITION
+
+enum Roles {
+  ADMIN
+  USER
+}
+
+type Query {
+  searchRole(roles: [Roles!]! = [ADMIN]): Int! @auth
+}
+```
+
+Add the option `customDirective` to the `docusaurus2-graphql-doc-generator` configuration.
+
+```js {6-19}
+plugins: [
+  [
+    "@graphql-markdown/docusaurus",
+    {
+      // ... other options
+      customDirective: {
+        auth: {
+          descriptor: (directive, type) => {
+            // get value for argument "requires"
+            const value = directiveDescriptor(
```

Comment:
small typo, we would return the `value`.

Reply:
Fixed in b92ce35